src/rebar_prv_update.erl

```diff
@@ -114,15 +114,11 @@ hex_to_index(State) ->
 
                 ets:foldl(fun({Pkg, [[]]}, _) when is_binary(Pkg) ->
                                   true;
-                             ({Pkg, [Vsns=[Vsn | _Rest]]}, _) when is_binary(Pkg) ->
+                             ({Pkg, [Vsns=[_Vsn | _Rest]]}, _) when is_binary(Pkg) ->
                                   %% Verify the package is of the right build tool by checking if the first
                                   %% version exists in the table from the foldl above
-                                  case ets:member(?PACKAGE_TABLE, {Pkg, Vsn}) of
-                                      true ->
-                                          ets:insert(?PACKAGE_TABLE, {Pkg, Vsns});
-                                      false ->
-                                          true
-                                  end;
+                                  Vsns1 = [V || V <- Vsns, ets:member(?PACKAGE_TABLE, {Pkg, V})],
+                                  ets:insert(?PACKAGE_TABLE, {Pkg, Vsns1});
```

Comment:
I figured that was fine/expected. But I can modify it to only insert if one matches.